packages/backend/src/nest/storage/UserProfileStore.ts

```diff
@@ -0,0 +1,261 @@
+import { EventEmitter } from 'events'
+import KeyValueStore from 'orbit-db-kvstore'
+import OrbitDB from 'orbit-db'
+import { getCrypto, ICryptoEngine } from 'pkijs'
+import { sha256 } from 'multiformats/hashes/sha2'
+import { stringToArrayBuffer } from 'pvutils'
+import * as Block from 'multiformats/block'
+import * as dagCbor from '@ipld/dag-cbor'
+
+import { Logger } from '@quiet/logger'
+import { NoCryptoEngineError, UserProfile } from '@quiet/types'
+import { keyObjectFromString, verifyDataSignature } from '@quiet/identity'
+
+import { StorageEvents } from './storage.types'
+import createLogger from '../common/logger'
+
+const logger = createLogger('UserProfileStore')
+
+/**
+ * Check magic byte sequence to determine if buffer is a PNG image.
+ */
+export const isPng = (buffer: Uint8Array): boolean => {
+  // https://en.wikipedia.org/wiki/PNG
+  const pngHeader = [0x89, 0x50, 0x4e, 0x47, 0x0d, 0x0a, 0x1a, 0x0a]
+
+  if (buffer.length < pngHeader.length) {
+    return false
+  }
+
+  for (let i = 0; i < pngHeader.length; i++) {
+    if (buffer[i] !== pngHeader[i]) {
+      return false
+    }
+  }
+  return true
+}
+
+/**
+ * Takes a base64 data URI string that starts with 'data:*\/*;base64,'
+ * as returned from
+ * https://developer.mozilla.org/en-US/docs/Web/API/FileReader/readAsDataURL
+ * and converts it to a Uint8Array.
+ */
+export const base64DataURLToByteArray = (contents: string): Uint8Array => {
+  const [header, base64Data] = contents.split(',')
+  if (!header.startsWith('data:') || !header.endsWith(';base64')) {
+    throw new Error('Expected base64 data URI')
+  }
+  const chars = atob(base64Data)
+  const bytes = new Array(chars.length)
+  for (let i = 0; i < chars.length; i++) {
+    bytes[i] = chars.charCodeAt(i)
+  }
+  return new Uint8Array(bytes)
+}
+
+export class UserProfileStore {
```

Comment:
In overall I will suggest to do it as injectable provider like another services.
https://docs.nestjs.com/providers

Then you have to import that in `providers` array in storage.module.ts and inject in StorageService

Some advantages of this approach

-We will have consistency between other services and will be consistent with the NestJS documentation
-in future You can inject via constructor another providers like options, or other services
-it will be easier to test that part of code in future with another services